### ISSUE TITLE
[bug]: Unable to generate test report

### ISSUE THREAD

**Issue by JammUtkarsh** [keploy/keploy#510]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When I try to record the test report of a kfasthttp application like [this one](https://github.com/JammUtkarsh/samples-go/tree/fasthttp-mysql/fasthttp-mysql). The test report is empty, since the `keploy/tests` and `keploy/mocks` are not being detected(maybe).

I am getting the following output:
```bash keploy                                                                                                                                                 9:13:20

       ▓██▓▄
    ▓▓▓▓██▓█▓▄
     ████████▓▒
          ▀▓▓███▄      ▄▄   ▄               ▌
         ▄▌▌▓▓████▄    ██ ▓█▀  ▄▌▀▄  ▓▓▌▄   ▓█  ▄▌▓▓▌▄ ▌▌   ▓
       ▓█████████▌▓▓   ██▓█▄  ▓█▄▓▓ ▐█▌  ██ ▓█  █▌  ██  █▌ █▓
      ▓▓▓▓▀▀▀▀▓▓▓▓▓▓▌  ██  █▓  ▓▌▄▄ ▐█▓▄▓█▀ █▓█ ▀█▄▄█▀   █▓█
       ▓▌                           ▐█▌                   █▌
        ▓

keploy 0.8.1

.2023-03-27T09:13:39.782+0530	INFO	server/server.go:215	keploy started at port 6789
2023-03-27T09:13:42.548+0530	INFO	testCase/testCase.go:128	no testcases found in /Users/utc/code/openSource/samples-go/fasthttp-mysql/keploy/tests directory.
2023-03-27T09:13:42.562+0530	INFO	testCase/testCase.go:228
💾 Recorded testcase with name: test-1 in yaml file at path: /Users/utc/code/openSource/samples-go/fasthttp-mysql/keploy/tests

2023-03-27T09:13:42.564+0530	INFO	testCase/testCase.go:238
💾 Recorded mocks for testcase with name: test-1 at path: /Users/utc/code/openSource/samples-go/fasthttp-mysql/keploy/mocks

2023-03-27T09:14:14.660+0530	INFO	testCase/testCase.go:228
💾 Recorded testcase with name: test-2 in yaml file at path: /Users/utc/code/openSource/samples-go/fasthttp-mysql/keploy/tests

2023-03-27T09:14:14.661+0530	INFO	testCase/testCase.go:238
💾 Recorded mocks for testcase with name: test-2 at path: /Users/utc/code/openSource/samples-go/fasthttp-mysql/keploy/mocks

2023-03-27T09:14:30.059+0530	INFO	testCase/testCase.go:228
💾 Recorded testcase with name: test-3 in yaml file at path: /Users/utc/code/openSource/samples-go/fasthttp-mysql/keploy/tests

2023-03-27T09:14:30.060+0530	INFO	testCase/testCase.go:238
💾 Recorded mocks for testcase with name: test-3 at path: /Users/utc/code/openSource/samples-go/fasthttp-mysql/keploy/mocks


 <=========================================>
  TESTRUN STARTED with id: "d2db0330-5b69-43c4-99c5-8a11f587c759"
	For App: "fasthttp-mysql"
	Total tests: 0
 <=========================================>

2023-03-27T09:15:30.919+0530	ERROR	regression/regression.go:90	found no test results for test report with id: d2db0330-5b69-43c4-99c5-8a11f587c759
go.keploy.io/server/pkg/service/regression.(*Regression).stopTestRun
	/Users/runner/work/keploy/keploy/pkg/service/regression/regression.go:90
go.keploy.io/server/pkg/service/regression.(*Regression).PutTest
	/Users/runner/work/keploy/keploy/pkg/service/regression/regression.go:898
go.keploy.io/server/http/regression.(*regression).End
	/Users/runner/work/keploy/keploy/http/regression/regression.go:69
net/http.HandlerFunc.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2047
github.com/go-chi/chi.(*Mux).routeHTTP
	/Users/runner/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:436
net/http.HandlerFunc.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2047
github.com/go-chi/chi.(*Mux).ServeHTTP
	/Users/runner/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:70
github.com/go-chi/chi.(*Mux).Mount.func1
	/Users/runner/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:311
net/http.HandlerFunc.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2047
github.com/go-chi/chi.(*Mux).routeHTTP
	/Users/runner/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:436
net/http.HandlerFunc.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2047
github.com/go-chi/chi.(*Mux).ServeHTTP
	/Users/runner/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:70
github.com/go-chi/chi.(*Mux).Mount.func1
	/Users/runner/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:311
net/http.HandlerFunc.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2047
github.com/go-chi/chi.(*Mux).routeHTTP
	/Users/runner/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:436
net/http.HandlerFunc.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2047
github.com/go-chi/cors.(*Cors).Handler.func1
	/Users/runner/go/pkg/mod/github.com/go-chi/cors@v1.2.0/cors.go:228
net/http.HandlerFunc.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2047
github.com/keploy/go-sdk/integrations/kchi.(*chiV5).Next
	/Users/runner/go/pkg/mod/github.com/keploy/go-sdk@v0.7.9/integrations/kchi/chi-v5.go:64
github.com/keploy/go-sdk/keploy.Middleware
	/Users/runner/go/pkg/mod/github.com/keploy/go-sdk@v0.7.9/keploy/router.go:34
github.com/keploy/go-sdk/integrations/kchi.ChiMiddlewareV5.func1.1
	/Users/runner/go/pkg/mod/github.com/keploy/go-sdk@v0.7.9/integrations/kchi/chi-v5.go:13
net/http.HandlerFunc.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2047
github.com/go-chi/chi.(*Mux).ServeHTTP
	/Users/runner/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:87
net/http.serverHandler.ServeHTTP
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:2879
net/http.(*conn).serve
	/Users/runner/hostedtoolcache/go/1.17.13/x64/src/net/http/server.go:1930

 <=========================================>
  TESTRUN SUMMARY. For testrun with id: "d2db0330-5b69-43c4-99c5-8a11f587c759"
	Total tests: 0
	Total test passed: 0
	Total test failed: 0
 <=========================================>

^C
```

A visual way of showing the above info is:
<img width="2077" alt="main.go" src="https://user-images.githubusercontent.com/78140690/227837824-d5f4a45e-70a8-4800-b9dc-f4643422d903.png">

<img width="2077" alt="handlers.go" src="https://user-images.githubusercontent.com/78140690/227837852-c0232c27-740b-4038-9922-84e70311cdf1.png">

### Steps to reproduce

Follow the steps from [sample-go/fasthttp-mysql](https://github.com/JammUtkarsh/samples-go/tree/fasthttp-mysql/fasthttp-mysql) application.

On the stage when you run `export KEPLOY_MODE=test` and `go test -v -coverpkg=./... -covermode=atomic  ./...` you will get the above text.

### Environment

None

### Version

Local

### Repository

Keploy

**Comment by JammUtkarsh** [comment]
Upon further debugging and inspecting, I found the following things:
**NOTE:** I though this issue was specific just on my machine. But I followed the instructions by book in [DIND](https://hub.docker.com/_/docker)(Docker INside Docker) container.

There are three operations
1. POST a movie
2. GET last movie added. Also also a sub SQL command of COUNT(ID) (Maybe this could be causing error)
3. GET all movies

If I run 1 and 3, it works perfectly.
When run 2 independently(or with 1 and 3). I am getting the following log
```bash
[
  {
    "level": "error",
    "ts": 1679939266.4119606,
    "caller": "keploy/utils.go:113",
    "msg": "dependency mocking failed: failed to decode object",
    "type": "nil",
    "test id": "test-2",
    "stacktrace": "github.com/keploy/go-sdk/keploy.ProcessDep\n\t/root/go/pkg/mod/github.com/keploy/go-sdk@v0.8.3/keploy/utils.go:113\ngithub.com/keploy/go-sdk/integrations/ksql/v1.Rows.Columns\n\t/root/go/pkg/mod/github.com/keploy/go-sdk@v0.8.3/integrations/ksql/v1/queryerContext.go:58\ndatabase/sql.(*Rows).nextLocked\n\t/usr/lib/go/src/database/sql/sql.go:3005\ndatabase/sql.(*Rows).Next.func1\n\t/usr/lib/go/src/database/sql/sql.go:2986\ndatabase/sql.withLock\n\t/usr/lib/go/src/database/sql/sql.go:3439\ndatabase/sql.(*Rows).Next\n\t/usr/lib/go/src/database/sql/sql.go:2985\ndatabase/sql.(*Row).Scan\n\t/usr/lib/go/src/database/sql/sql.go:3376\nfasthttp-sql/models.SingleMovie\n\t/root/samples-go/fasthttp-mysql/models/movies.go:25\nfasthttp-sql.GETmovie\n\t/root/samples-go/fasthttp-mysql/handlers.go:50\nfasthttp-sql.main.func1\n\t/root/samples-go/fasthttp-mysql/main.go:37\ngithub.com/keploy/go-sdk/integrations/kfasthttp.captureResp\n\t/root/go/pkg/mod/github.com/keploy/go-sdk@v0.8.3/integrations/kfasthttp/fasthttp.go:27\ngithub.com/keploy/go-sdk/integrations/kfasthttp.FastHttpMiddleware.func2.1\n\t/root/go/pkg/mod/github.com/keploy/go-sdk@v0.8.3/integrations/kfasthttp/fasthttp.go:65\ngithub.com/valyala/fasthttp.(*Server).serveConn\n\t/root/go/pkg/mod/github.com/valyala/fasthttp@v1.45.0/server.go:2371\ngithub.com/valyala/fasthttp.(*workerPool).workerFunc\n\t/root/go/pkg/mod/github.com/valyala/fasthttp@v1.45.0/workerpool.go:224\ngithub.com/valyala/fasthttp.(*workerPool).getCh.func1\n\t/root/go/pkg/mod/github.com/valyala/fasthttp@v1.45.0/workerpool.go:196"
  },
  {
    "level": "error",
    "ts": 1679939266.4125128,
    "caller": "keploy/utils.go:97",
    "msg": "mocking failed: Async or Unrecorded dependency call. Please record again and delete current tcs with",
    "test id": "test-2",
    "stacktrace": "github.com/keploy/go-sdk/keploy.ProcessDep\n\t/root/go/pkg/mod/github.com/keploy/go-sdk@v0.8.3/keploy/utils.go:97\ngithub.com/keploy/go-sdk/integrations/ksql/v1.Rows.Next\n\t/root/go/pkg/mod/github.com/keploy/go-sdk@v0.8.3/integrations/ksql/v1/queryerContext.go:261\ndatabase/sql.(*Rows).nextLocked\n\t/usr/lib/go/src/database/sql/sql.go:3008\ndatabase/sql.(*Rows).Next.func1\n\t/usr/lib/go/src/database/sql/sql.go:2986\ndatabase/sql.withLock\n\t/usr/lib/go/src/database/sql/sql.go:3439\ndatabase/sql.(*Rows).Next\n\t/usr/lib/go/src/database/sql/sql.go:2985\ndatabase/sql.(*Row).Scan\n\t/usr/lib/go/src/database/sql/sql.go:3376\nfasthttp-sql/models.SingleMovie\n\t/root/samples-go/fasthttp-mysql/models/movies.go:25\nfasthttp-sql.GETmovie\n\t/root/samples-go/fasthttp-mysql/handlers.go:50\nfasthttp-sql.main.func1\n\t/root/samples-go/fasthttp-mysql/main.go:37\ngithub.com/keploy/go-sdk/integrations/kfasthttp.captureResp\n\t/root/go/pkg/mod/github.com/keploy/go-sdk@v0.8.3/integrations/kfasthttp/fasthttp.go:27\ngithub.com/keploy/go-sdk/integrations/kfasthttp.FastHttpMiddleware.func2.1\n\t/root/go/pkg/mod/github.com/keploy/go-sdk@v0.8.3/integrations/kfasthttp/fasthttp.go:65\ngithub.com/valyala/fasthttp.(*Server).serveConn\n\t/root/go/pkg/mod/github.com/valyala/fasthttp@v1.45.0/server.go:2371\ngithub.com/valyala/fasthttp.(*workerPool).workerFunc\n\t/root/go/pkg/mod/github.com/valyala/fasthttp@v1.45.0/workerpool.go:224\ngithub.com/valyala/fasthttp.(*workerPool).getCh.func1\n\t/root/go/pkg/mod/github.com/valyala/fasthttp@v1.45.0/workerpool.go:196"
  }
]
```
and then exits with `panic()`

You may check if there is anything wrong in my implementation [here](https://github.com/JammUtkarsh/samples-go/blob/fasthttp-mysql/fasthttp-mysql/handlers.go#L51-L58). If there is anything wrong, please let me know!

A visual view:
<img width="1622" alt="Screenshot 2023-03-27 at 23 32 54" src="https://user-images.githubusercontent.com/78140690/229016832-22c6348a-9f12-480b-8406-86687644abfc.png">

**Comment by Sarthak160** [comment]
> 

@JammUtkarsh can you please send tests and mocks generated on slack . I will check it once .

**Comment by JammUtkarsh** [comment]
Thanks to @re-Tick

This was closed in [#183](https://github.com/keploy/go-sdk/pull/183)
